### PR TITLE
Add unoccupied column pool

### PR DESCRIPTION
**Pull request by ghost**
Adds an interface for the unoccupied column pool and a simple implementation that hands out columns in a round-robin fashion.  It is a little forgiving with certain erroneous usages (like `Restore()` ing columns that never existed), but it should stay in a good state even if it doesn't raise an exception.

### CODE REVIEW

**Review by JamesComo44 — APPROVED** [review]
My boi goes from hello world to using structs. <3

File: DigitalRain/RoundRobinUnoccupiedColumnPool.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                _columnPool[_nextId] = false;
+
+                var idToReturn = _nextId;
+                _nextId = Increment(_nextId);
+                return new ColumnId(idToReturn);
+            }
+            throw new InvalidOperationException("No unoccupied columns available in the pool");
+        }
+
+        public void Restore(ISet<ColumnId> columnsToRestore)
```
Comment (JamesComo44) [inline DigitalRain/RoundRobinUnoccupiedColumnPool.cs:27]: R E S T O R E

File: DigitalRain/RoundRobinUnoccupiedColumnPool.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            if (_columnPool[_nextId])
+            {
+                _columnPool[_nextId] = false;
+
+                var idToReturn = _nextId;
+                _nextId = Increment(_nextId);
+                return new ColumnId(idToReturn);
+            }
+            throw new InvalidOperationException("No unoccupied columns available in the pool");
```
Comment (JamesComo44) [inline DigitalRain/RoundRobinUnoccupiedColumnPool.cs:24]: Throw'em!

File: DigitalRain/RoundRobinUnoccupiedColumnPool.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            return (id + 1) % ColumnCount;
+        }
+
+        public int ColumnCount
+        {
+            get { return _columnPool.Length; }
+        }
+
+        private readonly bool[] _columnPool;
```
Comment (JamesComo44) [inline DigitalRain/RoundRobinUnoccupiedColumnPool.cs:48]: I see. You want to initialize private vars at the bottom of the file?
I can update my PR to reflect that.

File: Tests/UnitTest1.cs [hunk 4]
```diff
@@ -1,16 +0,0 @@
-using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (JamesComo44) [inline Tests/UnitTest1.cs:1]: "Hello for the last time?"

**Comment by JamesComo44** [comment]
Thinking about conding conventions. Should we allow the use of C#'s `var`¿

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/var

File: DigitalRain/RoundRobinUnoccupiedColumnPool.cs (hunk 3 above)
Comment (ghost) [reply DigitalRain/RoundRobinUnoccupiedColumnPool.cs:48]: 🤷‍♂️ i don't mind the order either way just as long as they're grouped

File: Tests/UnitTest1.cs (hunk 4 above)
Comment (ghost) [reply Tests/UnitTest1.cs:1]: It was the third attempt at a hello world test lol

**Comment by ghost** [comment]
> Thinking about conding conventions. Should we allow the use of C#'s `var`¿
> 
> https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/var

I love var. Is there a reason we shouldn't?
I like how much redundancy it reduces.  It kinda makes it feel more like Python with mypy's type inference.  But in Python you get it by default, and have to go out of your way to specify the type, whereas in C# you have to specify a type by default, but you can opt to have it inferred instead with var.

**Pull request merged**